superset/views/core.py

```diff
@@ -105,8 +106,12 @@
     get_viz,
 )
 
+
 config = app.config
 CACHE_DEFAULT_TIMEOUT = config.get("CACHE_DEFAULT_TIMEOUT", 0)
+SQLLAB_QUERY_COST_ESTIMATE_TIMEOUT = config.get(
```

Comment:
not a big fan of configuring different timeouts for each different API call, is there a reason this is needed?

Reply:
My impression is that here we want a shorter timeout than usual. The query should return quickly, and if it doesn't return in a few seconds there's probably something wrong and it's not worth waiting for.